src/pages/home/report/withReportAndReportActionOrNotFound.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {ComponentType, ForwardedRef, RefAttributes} from 'react';
+import {withOnyx, OnyxEntry} from 'react-native-onyx';
+import {RouteProp} from '@react-navigation/native';
+import isEmpty from 'lodash/isEmpty';
+import getComponentDisplayName from '../../../libs/getComponentDisplayName';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import ONYXKEYS from '../../../ONYXKEYS';
+import type {Report, Policy, Beta} from '../../../types/onyx';
+import FullscreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import * as ReportUtils from '../../../libs/ReportUtils';
+
+type OnyxProps = {
+    /** The report currently being looked at */
+    report: OnyxEntry<Report>;
+
+    /** The policies which the user has access to */
+    policies: OnyxEntry<Policy>;
+
+    /** Beta features list */
+    betas: OnyxEntry<Beta[]>;
+
+    /** Indicated whether the report data is loading */
+    isLoadingReportData: OnyxEntry<boolean>;
+};
+
+type ComponentProps = OnyxProps & {
+    route: RouteProp<{params: {reportID: string}}>;
+};
+
+export default function <TProps extends ComponentProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportOrNotFound(props: OnyxProps, ref: ForwardedRef<TRef>) {
+        const contentShown = React.useRef(false);
+
+        const shouldShowFullScreenLoadingIndicator = props.isLoadingReportData && (isEmpty(props.report) || !props.report.reportID);
```

Comment:
Maybe you can get rid of lodash `isEmpty` the same way as it was done in this PR for example https://github.com/Expensify/App/pull/29768/files

Reply:
removed